### PR TITLE
Improve Functionality and Experience By Patternfly-ng

### PR DESCRIPTION
**Pull request by rajatsingh25aug**
Sorting Buttons and Along with the avatar of the profile.Improved overall user experience by adding patternfly.

### CODE REVIEW

**Comment by rohitkrai03** [comment]
@rajatsingh25aug Don't commit package-lock.json changes if there is no package being installed or updated.

File: src/app/components/search-user/search-user.component.html [hunk 1]
```diff
@@ -3,10 +3,10 @@
...
     placeholder="Search"
-    id="searchbox"
     (keyup.enter)="onSearch(searchBar.value)"
+    class="form-control"
     />
-  <button
-    type="button" (click)="onSearch(searchBar.value)">Search
-  </button>
 </div>
+
```
Comment (rohitkrai03) [inline src/app/components/search-user/search-user.component.html]: There shouldn't be any unnecessary empty spaces

File: src/app/components/search-user/search-user.component.html [hunk 2]
```diff
@@ -3,10 +3,10 @@
   #searchBar
     type="text"
     placeholder="Search"
-    id="searchbox"
     (keyup.enter)="onSearch(searchBar.value)"
+    class="form-control"
     />
-  <button
```
Comment (rohitkrai03) [inline src/app/components/search-user/search-user.component.html]: Why did you remove the button?

File: src/app/components/show-user/show-user.component.html [hunk 3]
```diff
@@ -1,4 +1,114 @@
-<div *ngFor="let user of store.users | async">
-  <h3>{{ user.attributes.fullName }}</h3>
-  <h4>{{ user.attributes.email }}</h4>
+<!--navbar starts-->
+<div class="container-fluid" *ngIf="users">
```
Comment (rohitkrai03) [inline src/app/components/show-user/show-user.component.html]: Using `toolbar` and `list` components from `patternfly-ng` instead of directly using the HTML markup will. be better here. Maybe you can look into it in separate PR.

File: src/app/components/show-user/show-user.component.html [hunk 4]
```diff
@@ -1,4 +1,114 @@
...
+<li><a href="#">Action</a></li>
+<li><a href="#">Another Action</a></li>
+<li><a href="#"> Something Else Here</a></li>
+<li role="separator" class="divider"></li>
+<li><a href="#">Separated Link</a></li>
+</ul>
 </div>
+
+        </div>
+
```
Comment (rohitkrai03) [inline src/app/components/show-user/show-user.component.html]: The whole file has very wierd indentation. Fix that and empty spaces.

File: src/app/components/show-user/show-user.component.html [hunk 5]
```diff
@@ -1,4 +1,114 @@
...
+              <h3>{{user.attributes.fullName }}</h3>
+          </div>
+        </div>
+        <div class="list-view-pf-additional-info">
+          <div class="list-view-pf-additional-info-item">
+            <span class="pficon pficon-screen"></span>
+            <strong><h4>{{user.attributes.email }}</h4></strong>
+            <div style="margin: 20px 0;">
+                <a href="" class="btn btn-default" data-toggle="popover" data-html="true" title="" data-content="Empty Title example.
+                  This fis more information about a smartproxy. This is more information about a smartproxy. This is more information about a smartproxy.">Click To find Bio
```
Comment (rohitkrai03) [inline src/app/components/show-user/show-user.component.html]: What is the `data-content` here? It should be the actuaal `bio` of user.

**Comment by rajatsingh25aug** [comment]
Sure @rohitkrai03 .I never knew about that. Thanks for the info!.

File: src/app/components/show-user/show-user.component.html (hunk 3 above)
Comment (rajatsingh25aug) [reply src/app/components/show-user/show-user.component.html]: yes sure. I'll raise a pr when I'll make view profile component with angular-ng.

File: src/app/components/search-user/search-user.component.html (hunk 1 above)
Comment (rajatsingh25aug) [reply src/app/components/search-user/search-user.component.html]: Sorry,I'll fix all the unnecessary spaces and raise a pr again!. Linter didn't show me any error so I ignored that.

Comment (rohitkrai03) [reply src/app/components/search-user/search-user.component.html]: No need to raise another PR. Just add commits to your branch and push it. The PR will update automatically.

File: src/app/components/show-user/show-user.component.html (hunk 3 above)
Comment (rohitkrai03) [reply src/app/components/show-user/show-user.component.html]: angular-ng?

File: src/app/components/search-user/search-user.component.html (hunk 1 above)
Comment (rajatsingh25aug) [reply src/app/components/search-user/search-user.component.html]: Ok Sure

**Comment by rajatsingh25aug** [comment]
No idea why is it happening!.☹️

**Comment by rohitkrai03** [comment]
@rajatsingh25aug You will need to rebase your branch and resolve all the conflicts. Take help from @nemesis09

**Comment by rajatsingh25aug** [comment]
Please merge this.evrything works fine now!. The code is all clean, readable and working perfectly!.

**Comment by rajatsingh25aug** [comment]
Don't merge. Just don't merge rn

File: src/app/app.module.ts [hunk 6]
```diff
@@ -14,6 +14,10 @@ import { UsersContainerComponent } from './components/users-container/users-cont
 import { UsersListComponent } from './components/users-container/users-list/users-list.component';
 import { UsersDataStore } from './store/users-data.store';
 import { SearchBarComponent } from './components/search-bar/search-bar.component';
+import { ListModule } from 'patternfly-ng';
```
Comment (rohitkrai03) [inline src/app/app.module.ts]: You can merge all the imports into one line from `patternfly-ng`

File: src/app/app.module.ts [hunk 7]
```diff
@@ -43,6 +51,6 @@ import { SearchBarComponent } from './components/search-bar/search-bar.component
     { provide: REALM, useValue: 'realm' },
     UsersDataStore
   ],
-  bootstrap: [AppComponent]
+  bootstrap: [AppComponent],
```
Comment (rohitkrai03) [inline src/app/app.module.ts]: No empty commas at the end.

File: src/app/components/search-bar/search-bar.component.html [hunk 8]
```diff
@@ -1,12 +1,10 @@
...
+<div class="search-pf has-button Search">
+    <div class="form-group">
+        <div>
+            <input #searchBar type="text" (keyup.enter)="handleSearch(searchBar.value)" class="form-control" placeholder="Search" />
+        </div>
+    </div>
+    <div class="form-group">
+        <button class="btn btn-default" type="button" (click)="handleSearch(searchBar.value)"><span class="fa fa-search"></span></button>
+    </div>
+</div>
```
Comment (rohitkrai03) [inline src/app/components/search-bar/search-bar.component.html]: Needs empty line at end.

File: src/app/components/search-bar/search-bar.component.html [hunk 9]
```diff
@@ -1,12 +1,10 @@
...
-    type="text"
-    placeholder="Search"
-    id="searchBar"
-    (keyup.enter)="handleSearch(searchBar.value)"
-  />
-  <button
-    type="button" (click)="handleSearch(searchBar.value)"> Search
-  </button>
-</div>
+<div class="search-pf has-button Search">
```
Comment (rohitkrai03) [inline src/app/components/search-bar/search-bar.component.html]: This file has indentation with 4 spaces. It should be 2.

File: src/app/components/users-container/users-container.component.html [hunk 10]
```diff
@@ -1,7 +1,7 @@
 <app-search-bar
```
Comment (rohitkrai03) [inline src/app/components/users-container/users-container.component.html]: Same as above. Indent with 2 spaces.

File: src/app/components/users-container/users-list/users-list.component.html [hunk 11]
```diff
@@ -1,12 +1,54 @@
...
+                            <div class="list-pf-content-wrapper">
+                                <div class="list-pf-main-content">
+                                    <div class="list-pf-title">{{item.attributes.fullName}}</div>
+                                    <div class="list-pf-description text-overflow-pf">{{item.attributes.username}}</div>
+                                    <div class="list-pf-description text-overflow-pf">{{item.attributes.email}}</div>
+                                </div>
+                            </div>
+                        </ng-template>
+                        <ng-template #expandTemplate let-item="item" let-index="index">
+                            <li style="font-size:30px;"><span><strong>Company:</strong> {{item.attributes.created-at}}</span></li>
```
Comment (rohitkrai03) [inline src/app/components/users-container/users-list/users-list.component.html]: No changes to the comments i gave here before. Don't use inline css. don't use <strong>

File: src/app/components/users-container/users-list/users-list.component.html [hunk 12]
```diff
@@ -1,12 +1,54 @@
...
+                            <li style="font-size:30px;"><strong>Url:</strong><span *ngIf="item.attributes.url;else noUrl ">{{item.attributes.url}}</span></li>
+                        </ng-template>
+                    </pfng-list>
+                </div>
+            </div>
+        </div>
+    </div>
+</div>
+<!--Expansion Ends-->
+<ng-template #noBio>
```
Comment (rohitkrai03) [inline src/app/components/users-container/users-list/users-list.component.html]: Keep the templates where it is relevant

File: src/app/components/users-container/users-list/users-list.component.ts [hunk 13]
```diff
@@ -1,16 +1,142 @@
-import { Component, OnInit, Input } from '@angular/core';
+import { Component, OnInit, Input, SimpleChanges, OnChanges } from '@angular/core';
 import { User } from 'ngx-login-client';
-
+import { ListConfig } from 'patternfly-ng';
```
Comment (rohitkrai03) [inline src/app/components/users-container/users-list/users-list.component.ts]: Again so man y imports from same package on different lines.

File: src/polyfills.ts [hunk 14]
```diff
@@ -73,7 +73,7 @@ import 'core-js/es7/reflect';
  */
 import 'zone.js/dist/zone';  // Included with Angular CLI.
 
-
+(window as any).global = window;
```
[the remark this answers is not in the document]
Comment (rajatsingh25aug) [reply src/polyfills.ts]: it was throwing an error `global not defined`

[the remark this answers is not in the document]
Comment (rajatsingh25aug) [reply src/polyfills.ts]: we cannot remove this

File: src/app/components/users-container/users-list/users-list.component.ts [hunk 15]
```diff
@@ -1,16 +1,142 @@
...
-export class UsersListComponent implements OnInit {
+export class UsersListComponent implements OnInit, OnChanges {
 
   @Input() users: User[];
 
-  ngOnInit() {
+  listConfig: ListConfig;
+  filterConfig: FilterConfig;
+  filtersText: String = '';
+  items: any[];
```
Comment (rohitkrai03) [inline src/app/components/users-container/users-list/users-list.component.ts]: Why is items type `any[]`? It should be `User[]`.

File: src/app/app.module.ts [hunk 16]
```diff
@@ -1,5 +1,5 @@
 import { BrowserModule } from '@angular/platform-browser';
-import { NgModule } from '@angular/core';
+import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
```
[the remark this answers is not in the document]
Comment (rajatsingh25aug) [reply src/app/app.module.ts]: because then I can't use certain patternfly's tags.

File: src/app/app.module.ts (hunk 6 above)
Comment (rajatsingh25aug) [reply src/app/app.module.ts]: okay!

Comment (rajatsingh25aug) [reply src/app/app.module.ts]: sure

File: src/app/components/users-container/users-list/users-list.component.html (hunk 11 above)
Comment (rajatsingh25aug) [reply src/app/components/users-container/users-list/users-list.component.html]: okay sure!

File: src/app/components/users-container/users-list/users-list.component.html (hunk 12 above)
Comment (rajatsingh25aug) [reply src/app/components/users-container/users-list/users-list.component.html]: noBio is relevant. Several people don't have any bio.

File: src/app/components/users-container/users-list/users-list.component.ts (hunk 15 above)
Comment (rajatsingh25aug) [reply src/app/components/users-container/users-list/users-list.component.ts]: Forgot!.Will Fix it!.

File: src/app/components/users-container/users-list/users-list.component.html [hunk 17]
```diff
@@ -1,12 +1,130 @@
-<div *ngIf = "users">
-  <div *ngIf = "users.length; else noUser">
-  <div *ngFor="let user of users">
-    <h3>{{ user.attributes.fullName }}</h3>
-    <h4>{{ user.attributes.email }}</h4>
+<br>
```
[the remark this answers is not in the document]
Comment (rajatsingh25aug) [reply src/app/components/users-container/users-list/users-list.component.html]: yeah I will do that!

File: src/app/components/search-user/search-user.component.html (hunk 1 above)
Comment (rajatsingh25aug) [reply src/app/components/search-user/search-user.component.html]: components do not exist

File: src/app/components/search-user/search-user.component.html (hunk 2 above)
Comment (rajatsingh25aug) [reply src/app/components/search-user/search-user.component.html]: components do not exist

File: src/app/components/users-container/users-list/users-list.component.ts [hunk 18]
```diff
@@ -1,6 +1,24 @@
 import { Component, OnInit, Input } from '@angular/core';
 import { User } from 'ngx-login-client';
+import { ListConfig } from 'patternfly-ng';
+/*ToolBAR*/
 
+
+import { Action } from 'patternfly-ng';
```
[the remark this answers is not in the document]
Comment (rajatsingh25aug) [reply src/app/components/users-container/users-list/users-list.component.ts]: I'll merge it in one!

File: src/app/components/users-container/users-list/users-list.component.ts [hunk 19]
```diff
@@ -10,7 +28,114 @@ export class UsersListComponent implements OnInit {
...
+  items: any[];
+  isAscendingSort: Boolean = true;
+  separator: Object;
+  sortConfig: SortConfig;
+  currentSortField: SortField;
+  toolbarConfig: ToolbarConfig;
+  /*Config Files Ends */
+
+  ngOnInit(): void {
+    this.allItems = [{
```
[the remark this answers is not in the document]
Comment (rajatsingh25aug) [reply src/app/components/users-container/users-list/users-list.component.ts]: Nothing,I just removed it

File: src/app/components/users-container/users-list/users-list.component.ts [hunk 20]
```diff
@@ -10,7 +28,114 @@ export class UsersListComponent implements OnInit {
...
+      });
+    }
+  }
+
+   // Filter
+
+   applyFilters(filters: Filter[]): void {
+    this.items = [];
+    if (filters && filters.length > 0) {
+      this.allItems.forEach((item) => {
```
[the remark this answers is not in the document]
Comment (rajatsingh25aug) [reply src/app/components/users-container/users-list/users-list.component.ts]: removed

File: src/app/app.module.ts (hunk 7 above)
Comment (rajatsingh25aug) [reply src/app/app.module.ts]: i'll add

File: src/app/components/search-bar/search-bar.component.html (hunk 9 above)
Comment (rajatsingh25aug) [reply src/app/components/search-bar/search-bar.component.html]: Done
